core/src/main/java/org/apache/iceberg/actions/RewriteFileGroup.java

```diff
@@ -35,26 +35,32 @@
 public class RewriteFileGroup {
   private final FileGroupInfo info;
   private final List<FileScanTask> fileScanTasks;
+  private boolean isFailed;
 
   private Set<DataFile> addedFiles = Collections.emptySet();
 
-  public RewriteFileGroup(FileGroupInfo info, List<FileScanTask> fileScanTasks) {
+  public RewriteFileGroup(FileGroupInfo info, List<FileScanTask> fileScanTasks, boolean isFailed) {
```

Comment:
do we really need to pass a boolean here or could we derive that info maybe in a different way?

Reply:
It is better to set the default, thanks nastra.